src/Prototype/Exe/Form/Login.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE DeriveAnyClass #-}
+module Prototype.Exe.Form.Login
+  ( Input
+  , Page(..)
+  , ResultPage(..)
+  ) where
+
+import           Smart.Html.Dsl                 ( HtmlCanvas )
+import qualified Smart.Html.Dsl                as Dsl
+import qualified Smart.Html.Render             as Render
+import           Smart.Html.Shared.Html.Icons
+import           Text.Blaze                     ( customAttribute )
+import qualified Text.Blaze.Html5              as H
+import           Text.Blaze.Html5               ( (!) )
+import qualified Text.Blaze.Html5.Attributes   as A
+import           Web.FormUrlEncoded             ( FromForm(..)
+                                                , parseUnique
+                                                )
+
+
+--------------------------------------------------------------------------------
+data Input = Input
+  { username :: Text
+  , password :: Text
+  }
+  deriving (Eq, Show)
+
+instance FromForm Input where
+  fromForm f = Input <$> parseUnique "username" f <*> parseUnique "password" f
```

Comment:
We can use Generics to do this for us, no? 
```suggestion
data Input = Input
  { username :: Text
  , password :: Text
  }
  deriving (Eq, Show, G,eneric)
  deriving anyclass FromForm -- Will need DeriveAnyClass enabled. 

```

I'd also use `Password` from the `User` module instead of `Text` for the Password field.